Pod/Classes/LSTracer.m

```diff
@@ -373,23 +386,23 @@ - (void)flush:(void (^)(NSError * _Nullable error))doneCallback {
         req.timestampOffsetMicros = m_clockState.offsetMicros;
         m_pendingProtoSpans = [NSMutableArray<LSPBSpan*> array];
         m_lastFlush = now;
-        
+
         m_bgTaskId = [[UIApplication sharedApplication]
                       beginBackgroundTaskWithName:@"com.lightstep.flush"
                       expirationHandler:^{
-                          cleanupBlock([NSError errorWithDomain:LSErrorDomain code:LSBackgroundTaskError userInfo:nil]);
+                          cleanupBlock(true, [NSError errorWithDomain:LSErrorDomain code:LSBackgroundTaskError userInfo:nil]);
                       }];
         if (m_bgTaskId == UIBackgroundTaskInvalid) {
             NSLog(@"unable to enter the background, so skipping flush");
-            cleanupBlock([NSError errorWithDomain:LSErrorDomain code:LSBackgroundTaskError userInfo:nil]);
+            cleanupBlock(false, [NSError errorWithDomain:LSErrorDomain code:LSBackgroundTaskError userInfo:nil]);
```

Comment:
Since there's nothing to cancel in this case, we can avoid the deadlock possibility entirely.